apps/ejabberd/include/mongoose_commands.hrl

```diff
@@ -0,0 +1,32 @@
+
+-type command_action() :: [create|read|update|delete]. %% just basic CRUD; sending a mesage is 'create'
+
+-type typedef() :: integer|binary|float. %% most basic primitives, string is a binary
+
+-type argspec() :: typedef()
+                  |{atom(), typedef()} %% a named argument
+                  |{argspec()} % a tuple of a few args (can be of any size)
+                  |[typedef()]. % a list, but one element
+
+-type security() :: [admin|user]. %% later acl option will be added
+
+-type errortype() :: denied|not_implemented|type_error|internal. %% we should agree on a set of atoms so that the
+                                                                 %% frontend can map it to http codes
+
+-type failure() :: {error, errortype(), binary()}.
+
+-record(mongoose_command, {
```

Comment:
The record is used only by `mongoose_commands` module and tests. With some effort we can move it to mongoose_commands and remove the .hrl file completely. I don't think there is a need to access this record from tests.